controllers/machine_controller.go

```diff
@@ -270,20 +289,38 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	}
 
 	if isDeleteNodeAllowed {
-		// Drain node before deletion.
+		// Drain node before deletion and issue a patch in order to make this operation visible to the users.
 		if _, exists := m.ObjectMeta.Annotations[clusterv1.ExcludeNodeDrainingAnnotation]; !exists {
+			patchHelper, err := patch.NewHelper(m, r.Client)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
 			logger.Info("Draining node", "node", m.Status.NodeRef.Name)
+			conditions.MarkFalse(m, "DrainingSucceeded", "Draining", clusterv1.ConditionSeverityInfo, "Draining the node before deletion")
```

Comment:
Should "DrainingSucceeded" and "Draining" be constants?

Reply:
done!